Several video drivers failed to build against xlibre-xserver 25.1.6

### Select the version

25.1.X

### Describe your issue

I think it is more convenient to report it here than all the affected driver pages.

In my test, 6 drivers failed to build against xlibre-xserver 25.1.6 but successfully build against xlibre-xserver 25.1.5:
```
xf86-video-amdgpu-25.1.1
xf86-video-vmware-25.0.0
xf86-video-qxl-25.0.0
xf86-video-ati-25.0.1
xf86-video-nouveau-25.0.0
xf86-video-intel-25.0.2
```

Common error messages:
`error: static declaration of 'pci_device_is_boot_display' follows non-static declaration`
`note: previous declaration of 'pci_device_is_boot_display' with type 'int(struct pci_device *)'`




### Steps to reproduce

1. Update xlibre-xserver from 25.1.5 to 25.1.6
2. Rebuild all the drivers against xlibre-xserver

### What did you expect?

All build success

### Additional Information

Corresponding build logs of drivers:
```log
error: failed to build attribute 'checks.x86_64-linux.xlibre-xf86-video-amdgpu', build of '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-amdgpu-25.1.1.drv^*' failed: Cannot build '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-amdgpu-25.1.1.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-amdgpu-25.1.1
       Last 25 log lines:
       >    98 | static inline Bool pci_device_is_boot_display(struct pci_device *dev)
       >       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86Pci.h:108,
       >                  from ../src/amdgpu_drv.h:51:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libpciaccess-0.19/include/pciaccess.h:83:5: note: previous declaration of 'pci_device_is_boot_display' with type 'int(struct pci_device *)'
       >    83 | int pci_device_is_boot_display(struct pci_device *dev);
       >       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > [12/15] Compiling C object src/amdgpu_drv.so.p/amdgpu_dri3.c.o
       > FAILED: [code=1] src/amdgpu_drv.so.p/amdgpu_dri3.c.o
       > gcc -Isrc/amdgpu_drv.so.p -Isrc -I../src -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libdrm-2.4.133-dev/include -I. -I.. -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xorgproto-2025.1/include -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-mesa-libgbm-26.0.3/include -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libdrm-2.4.133-dev/include/libdrm -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xorgproto-2025.1/include/X11/dri -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-systemd-minimal-libs-260.1-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -fPIC -fvisibility=hidden -DHAVE_AMD_FMT_MOD_TILE_GFX12 -MD -MQ src/amdgpu_drv.so.p/amdgpu_dri3.c.o -MF src/amdgpu_drv.so.p/amdgpu_dri3.c.o.d -o src/amdgpu_drv.so.p/amdgpu_dri3.c.o -c ../src/amdgpu_dri3.c
       > In file included from ../src/amdgpu_probe.h:48,
       >                  from ../src/amdgpu_drv.h:64,
       >                  from ../src/amdgpu_dri3.c:26:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86platformBus.h:98:20: error: static declaration of 'pci_device_is_boot_display' follows non-static declaration
       >    98 | static inline Bool pci_device_is_boot_display(struct pci_device *dev)
       >       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86Pci.h:108,
       >                  from ../src/amdgpu_drv.h:51:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libpciaccess-0.19/include/pciaccess.h:83:5: note: previous declaration of 'pci_device_is_boot_display' with type 'int(struct pci_device *)'
       >    83 | int pci_device_is_boot_display(struct pci_device *dev);
       >       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > ../src/amdgpu_dri3.c:929:1: warning: 'amdgpu_dri3_get_drawable_modifiers' defined but not used [-Wunused-function]
       >   929 | amdgpu_dri3_get_drawable_modifiers(DrawablePtr draw, uint32_t format,
       >       | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > ninja: build stopped: subcommand failed.
       For full logs, run:
         nix log /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-amdgpu-25.1.1.drv
error: failed to build attribute 'checks.x86_64-linux.xlibre-xf86-video-vmware', build of '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-vmware-25.0.0.drv^*' failed: Cannot build '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-vmware-25.0.0.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-vmware-25.0.0
       Last 25 log lines:
       > make[2]: Leaving directory '/build/xf86-video-vmware-xlibre-xf86-video-vmware-25.0.0/vmwgfx'
       > Making all in src
       > make[2]: Entering directory '/build/xf86-video-vmware-xlibre-xf86-video-vmware-25.0.0/src'
       >   CC       vmware_drv_la-bits2pixels.lo
       >   CC       vmware_drv_la-vmware.lo
       >   CC       vmware_drv_la-vmwarecurs.lo
       >   CC       vmware_drv_la-vmwarectrl.lo
       >   CC       vmware_drv_la-vmwarexinerama.lo
       >   CC       vmware_drv_la-vmwarevideo.lo
       >   CC       vmware_drv_la-vmwaremodes.lo
       >   CC       vmware_drv_la-vmware_bootstrap.lo
       > In file included from vmware_bootstrap.c:38:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86platformBus.h:98:20: error: static declaration of 'pci_device_is_boot_display' follows non-static declaration
       >    98 | static inline Bool pci_device_is_boot_display(struct pci_device *dev)
       >       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86.h:70,
       >                  from vmware_bootstrap.c:30:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libpciaccess-0.19/include/pciaccess.h:83:5: note: previous declaration of 'pci_device_is_boot_display' with type 'int(struct pci_device *)'
       >    83 | int pci_device_is_boot_display(struct pci_device *dev);
       >       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[2]: *** [Makefile:598: vmware_drv_la-vmware_bootstrap.lo] Error 1
       > make[2]: Leaving directory '/build/xf86-video-vmware-xlibre-xf86-video-vmware-25.0.0/src'
       > make[1]: *** [Makefile:461: all-recursive] Error 1
       > make[1]: Leaving directory '/build/xf86-video-vmware-xlibre-xf86-video-vmware-25.0.0'
       > make: *** [Makefile:393: all] Error 2
       For full logs, run:
         nix log /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-vmware-25.0.0.drv
error: failed to build attribute 'checks.x86_64-linux.xlibre-xf86-video-qxl', build of '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-qxl-25.0.0.drv^*' failed: Cannot build '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-qxl-25.0.0.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-qxl-25.0.0
       Last 25 log lines:
       > Making all in spiceccid
       > make[3]: Entering directory '/build/xf86-video-qxl-xlibre-xf86-video-qxl-25.0.0/src/spiceccid'
       > make[3]: Nothing to be done for 'all'.
       > make[3]: Leaving directory '/build/xf86-video-qxl-xlibre-xf86-video-qxl-25.0.0/src/spiceccid'
       > make[3]: Entering directory '/build/xf86-video-qxl-xlibre-xf86-video-qxl-25.0.0/src'
       >   CC       qxl_driver.lo
       > In file included from qxl.h:49,
       >                  from qxl_driver.c:42:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86platformBus.h:98:20: error: static declaration of 'pci_device_is_boot_display' follows non-static declaration
       >    98 | static inline Bool pci_device_is_boot_display(struct pci_device *dev)
       >       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86.h:70,
       >                  from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86Modes.h:35,
       >                  from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86Crtc.h:28,
       >                  from qxl_driver.c:39:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libpciaccess-0.19/include/pciaccess.h:83:5: note: previous declaration of 'pci_device_is_boot_display' with type 'int(struct pci_device *)'
       >    83 | int pci_device_is_boot_display(struct pci_device *dev);
       >       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[3]: *** [Makefile:798: qxl_driver.lo] Error 1
       > make[3]: Leaving directory '/build/xf86-video-qxl-xlibre-xf86-video-qxl-25.0.0/src'
       > make[2]: *** [Makefile:988: all-recursive] Error 1
       > make[2]: Leaving directory '/build/xf86-video-qxl-xlibre-xf86-video-qxl-25.0.0/src'
       > make[1]: *** [Makefile:469: all-recursive] Error 1
       > make[1]: Leaving directory '/build/xf86-video-qxl-xlibre-xf86-video-qxl-25.0.0'
       > make: *** [Makefile:401: all] Error 2
       For full logs, run:
         nix log /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-qxl-25.0.0.drv
error: failed to build attribute 'checks.x86_64-linux.xlibre-xf86-video-ati', build of '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-ati-25.0.1.drv^*' failed: Cannot build '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-ati-25.0.1.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-ati-25.0.1
       Last 25 log lines:
       > build flags: SHELL=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bash-5.3p9/bin/bash
       > make  all-recursive
       > make[1]: Entering directory '/build/xf86-video-ati-xlibre-xf86-video-ati-25.0.1'
       > Making all in src
       > make[2]: Entering directory '/build/xf86-video-ati-xlibre-xf86-video-ati-25.0.1/src'
       >   CC       ati.lo
       >   CC       atimodule.lo
       >   CCLD     ati_drv.la
       >   CC       radeon_accel.lo
       > In file included from radeon_probe.h:48,
       >                  from radeon.h:63,
       >                  from radeon_accel.c:77:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86platformBus.h:98:20: error: static declaration of 'pci_device_is_boot_display' follows non-static declaration
       >    98 | static inline Bool pci_device_is_boot_display(struct pci_device *dev)
       >       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86Pci.h:108,
       >                  from radeon.h:50:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libpciaccess-0.19/include/pciaccess.h:83:5: note: previous declaration of 'pci_device_is_boot_display' with type 'int(struct pci_device *)'
       >    83 | int pci_device_is_boot_display(struct pci_device *dev);
       >       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[2]: *** [Makefile:692: radeon_accel.lo] Error 1
       > make[2]: Leaving directory '/build/xf86-video-ati-xlibre-xf86-video-ati-25.0.1/src'
       > make[1]: *** [Makefile:473: all-recursive] Error 1
       > make[1]: Leaving directory '/build/xf86-video-ati-xlibre-xf86-video-ati-25.0.1'
       > make: *** [Makefile:405: all] Error 2
       For full logs, run:
         nix log /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-ati-25.0.1.drv
error: failed to build attribute 'checks.x86_64-linux.xlibre-xf86-video-nouveau', build of '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-nouveau-25.0.0.drv^*' failed: Cannot build '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-nouveau-25.0.0.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-nouveau-25.0.0
       Last 25 log lines:
       >         Run 'make' to build xf86-video-nouveau
       >
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bash-5.3p9/bin/bash
       > make  all-recursive
       > make[1]: Entering directory '/build/xf86-video-nouveau-xlibre-xf86-video-nouveau-25.0.0'
       > Making all in src
       > make[2]: Entering directory '/build/xf86-video-nouveau-xlibre-xf86-video-nouveau-25.0.0/src'
       >   CC       nouveau_copy.lo
       > In file included from nv_include.h:22,
       >                  from nouveau_copy.h:4,
       >                  from nouveau_copy.c:25:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86platformBus.h:98:20: error: static declaration of 'pci_device_is_boot_display' follows non-static declaration
       >    98 | static inline Bool pci_device_is_boot_display(struct pci_device *dev)
       >       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86.h:70,
       >                  from nv_include.h:13:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libpciaccess-0.19/include/pciaccess.h:83:5: note: previous declaration of 'pci_device_is_boot_display' with type 'int(struct pci_device *)'
       >    83 | int pci_device_is_boot_display(struct pci_device *dev);
       >       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[2]: *** [Makefile:674: nouveau_copy.lo] Error 1
       > make[2]: Leaving directory '/build/xf86-video-nouveau-xlibre-xf86-video-nouveau-25.0.0/src'
       > make[1]: *** [Makefile:464: all-recursive] Error 1
       > make[1]: Leaving directory '/build/xf86-video-nouveau-xlibre-xf86-video-nouveau-25.0.0'
       > make: *** [Makefile:396: all] Error 2
       For full logs, run:
         nix log /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-nouveau-25.0.0.drv
error: failed to build attribute 'checks.x86_64-linux.xlibre-xf86-video-intel', build of '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-intel-25.0.2.drv^*' failed: Cannot build '/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-intel-25.0.2.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-intel-25.0.2
       Last 25 log lines:
       >   CC       intel_dri3.lo
       >   CC       intel_sync.lo
       >   CC       intel_present.lo
       >   CC       intel_hwmc.lo
       >   CCLD     libuxa.la
       > make[3]: Leaving directory '/build/xf86-video-intel-xlibre-xf86-video-intel-25.0.2/src/uxa'
       > make[3]: Entering directory '/build/xf86-video-intel-xlibre-xf86-video-intel-25.0.2/src'
       >   CC       backlight.lo
       >   CC       fd.lo
       >   CC       intel_device.lo
       > In file included from intel_device.c:54:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xlibre-xserver-25.1.6-dev/include/xorg/xf86platformBus.h:98:20: error: static declaration of 'pci_device_is_boot_display' follows non-static declaration
       >    98 | static inline Bool pci_device_is_boot_display(struct pci_device *dev)
       >       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > In file included from intel_device.c:44:
       > /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libpciaccess-0.19/include/pciaccess.h:83:5: note: previous declaration of 'pci_device_is_boot_display' with type 'int(struct pci_device *)'
       >    83 | int pci_device_is_boot_display(struct pci_device *dev);
       >       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[3]: *** [Makefile:609: intel_device.lo] Error 1
       > make[3]: Leaving directory '/build/xf86-video-intel-xlibre-xf86-video-intel-25.0.2/src'
       > make[2]: *** [Makefile:624: all-recursive] Error 1
       > make[2]: Leaving directory '/build/xf86-video-intel-xlibre-xf86-video-intel-25.0.2/src'
       > make[1]: *** [Makefile:498: all-recursive] Error 1
       > make[1]: Leaving directory '/build/xf86-video-intel-xlibre-xf86-video-intel-25.0.2'
       > make: *** [Makefile:430: all] Error 2
       For full logs, run:
         nix log /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xf86-video-intel-25.0.2.drv
error: some errors were encountered during the evaluation
```

### Extra fields

- [x] I have checked the existing [issues](https://github.com/X11Libre/xserver/issues)
- [x] I have read the [Contributing Guidelines](https://github.com/X11Libre/xserver/blob/master/CONTRIBUTING.md)
- [x] I'd like to work on this issue

--- a/hw/xfree86/xlibre-server.h.meson.in
+++ b/hw/xfree86/xlibre-server.h.meson.in
@@ -178,6 +178,9 @@
 /* Use libpciaccess */
 #mesondefine XSERVER_LIBPCIACCESS
 
+/* Have the pci_device_is_boot_display function */
+#mesondefine HAVE_PCI_DEVICE_IS_BOOT_DISPLAY
+
 /* X Access Control Extension */
 #mesondefine XACE
 

